### PR TITLE
feat(constants): Dependency Constants

### PR DESCRIPTION
**Pull request by souravdasslg**
1. Create dependency constant file
2. Export constants
3. Replace all dependency constants

#5088

### CODE REVIEW

File: lib/constants/dependency.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Dependency Type
+export const DEP_TYPE_CONTAINER_PULL = 'container_pull';
+export const DEP_TYPE_DEPENDENCY = 'dependencies';
```
Comment (JamieMagee) [inline lib/constants/dependency.ts]: Not 100% on the naming of this. Same for `devDependencies`, `peerDependencies`, and `optionalDependencies`.

Comment (rarkins) [reply lib/constants/dependency.ts]: Yeah, I was also unsure. In that case, spell them out? e.g. `DEP_TYPE_DEPENDENCIES`, etc?

Comment (JamieMagee) [reply lib/constants/dependency.ts]: Probably the best option.

**Review by rarkins — CHANGES_REQUESTED** [review]
Expand constants as mentioned in comments

**Review by rarkins — CHANGES_REQUESTED** [review]
Resolve conflicts

**Comment by rarkins** [comment]
I'm still formulating an opinion on the future of `depTypes`. Although this kind of constants formalization doesn't move things backwards, we may want to do a redesign of depTypes.

On one hand: it would be good if we fully documented our list of depTypes *where possible*, and per manager. This is because people may use them in `packageRules` so ideally they would have a way of knowing the possible values of depTypes generated.

On the other hand: there are package managers we have where the `depType` values can't be known ahead of time, because they are free text within package files (Bundler is an example). So we cannot ever enforce a list of valid depTypes within `packageRules`, even if to a human it's obvious (e.g. a typo like "dpendencies").

I wonder if each manager could export a value that indicates what type of depTypes it uses, e.g.

- `null` if it doesn't use them, or
- ["string", "array"] if it has a fixed set of values
- `"variable"` if it outputs them but they can't be predicted ahead of time

Such an export per-manager could be used as part of our autogenerated documentation.

The question would then be how to *enforce* it, so that our documentation doesn't become incorrect. It could perhaps be done in `datasource/index.ts` with this logic:
- if manager declares `null` but returns a depType or depTypes, issue a warning
- if manager declares an array but returns something outside of it, issue a warning

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconflicting

**Comment by rarkins** [comment]
I'm going to close this for now as we've done a lot of refactoring and may not need most of these.

**Pull request closed without merging**